### DISCUSSION TITLE
Have links across distinct files within a project work across output formats?

### DISCUSSION THREAD

**Discussion by reagle** [jgm/pandoc#10220] [Q&A] [answered]
I'm writing a book as separate markdown files, and want to refer to a table in chapter-2.md from chapter-3.md. I want to the link to the table to work independent of the output format (e.g., html, docx, odt). Is that possible by some semi-standard means?

That is, the following markdown:

```markdown
However, just as some judgement subs merit inclusion in the advice genre 
because they  perform a function of traditional columnists, 
see [Table 2.1](02-context.md#tbl_2_1_types-letters-subs), so do some disclosure subreddits.
```

when converted to html (or other format) should link to the HTML equivalent:

```html
<p>However, just as some judgement subs merit inclusion in the advice genre 
because they perform a function of traditional columnists, 
see <a href="02-context.html#tbl_2_1_types-letters-subs">Table 2.1</a>, so do some disclosure subreddits.</p>
```

**Comment by jgm** [comment]
You can do it with a filter. There's an example of this in one of the recent discussions or issues.

**Reply by reagle** [reply]
Thank you, but I [did look](https://github.com/jgm/pandoc/discussions?discussions_q=filter+to+convert+links+based+on+output+format) and didn't see anything. Pointers are welcome.

**Comment by jgm** [comment, marked as answer]
https://github.com/jgm/pandoc/issues/6384#issuecomment-2366834307

**Reply by reagle** [reply]
Ah. I did see #2719 (and noted it's about compiling into a single document) and #6834 (which appears similar) whereas I am speaking of maintaining separate input and output chapters wherein the links work between them. Additionally, the filters in #6834 seem to require that the output format extension be specified within the filter itself. I'm interested in something that somehow takes the output format and resulting extension from the command line's argument---so it works across whatever format I specify. Specifically, I want to maintain markdown links for ease of editing, HTML links when I'm reviewing, and DOCX for submitting to the publisher.

If this does requires a filter, those examples could be a start if combined with the use of the output-format argument...? (I didn't know if some crossref utility could do this or if pandoc had acquired a sense of multi-file projects.)

**Reply by jgm** [reply]
> the filters in https://github.com/jgm/pandoc/pull/6834 seem to require that the output format extension be specified within the filter itself.

Filters have access to the output file path via `PANDOC_STATE.output_file`, so you could just get the extension from there.
